src/poetry/vcs/git/backend.py

```diff
@@ -321,23 +321,44 @@ def _clone(cls, url: str, refspec: GitRefSpec, target: Path) -> Repo:
         return local
 
     @classmethod
-    def _clone_submodules(cls, repo: Repo) -> None:
+    def _clone_submodules(cls, repo: Repo, main_url: str) -> None:
         """
         Helper method to identify configured submodules and clone them recursively.
         """
         repo_root = Path(repo.path)
         modules_config = repo_root.joinpath(".gitmodules")
 
+        # A relative URL by definition starts with ../ or ./
+        relative_submodule_regex = re.compile("^(\\.\\./|\\./),?")
+
         if modules_config.exists():
             config = ConfigFile.from_path(str(modules_config))
 
             url: bytes
             path: bytes
             submodules = parse_submodules(config)
+
             for path, url, name in submodules:
                 path_relative = Path(path.decode("utf-8"))
                 path_absolute = repo_root.joinpath(path_relative)
 
+                url_string = url.decode()
```

Comment:
Thinking out loud here. Can't we simplify all by just doing something like this?

```py
from urllib.parse import urljoin

url_string = urljoin(f"{Git.get_remote_url(repo)}/", url.decode("utf-8"))
```
What I am not sure of is if `./foo.git` is ever a valid submodule. I am assuming here that git allows `./` only to enable relative paths such as `./../foo.git`. Because the above, if used with `./foo.git` in a superproject `https://github.com/baz/bar.git` will result in `https://github.com/baz/bar.git/foo.git`.

Reply:
This is more along the lines of what I was thinking, and would handle more cases... Tests for `./` and `./..` would be good to have as well (I for one have used `./` in a real project).

Reply:
@neersighted I tried to get `./foo.git` working on a project but event the git client does not like it when trying to do `git submodule init`. Can you share an example?

Reply:
Ah, I just realized the example I remembered uses `/proc/self/cwd` -- in other words, Linux-specific hacks.

Reply:
Implemented these changes locally. @neersighted I've also raised a [PR](https://github.com/python-poetry/test-fixture-vcs-repository/pull/6) in the test fixtures repo for the `./../foo.git` testcase. I'll update this PR when that one goes through.

Reply:
@neersighted Bumping this

Reply:
Resolving as I implemented this change and tests pass.